Arrangement_on_surface_2/doc/Arrangement_on_surface_2/CGAL/CORE_algebraic_number_traits.h

```diff
@@ -0,0 +1,218 @@
+namespace CGAL {
+
+/*! \ingroup PkgArrangementOnSurface2TraitsClasses
+ *
+ * `CORE_algebraic_number_traits` is a traits class for CORE's algebraic
+ * number types.
+ *
+ * \sa `Arr_conic_traits_2<RatKernel, AlgKernel, NtTraits>`
+ */
+class CORE_algebraic_number_traits {
+public:
+  /// \name Types
+  /// @{
+
+  //! The integer number type.
+  typedef CORE::BigInt                    Integer;
+
+  //! The rational number type.
+  typedef CORE::BigRat                    Rational;
+
+  //! The polynomial type.
+  typedef CORE::Polynomial<Integer>       Polynomial;
+
+  //! The algebraic number type.
+  typedef CORE::Expr                      Algebraic;
+
+  /// @}
+
+  /// \name Utility Functions
+  /// @{
+
+  /*! Obtain the numerator of a rational number.
+   * \param q A rational number.
+   * \return The numerator of q.
+   */
+  Integer numerator(const Rational& q) const;
+
+  /*! Obtain the denominator of a rational number.
+   * \param q A rational number.
+   * \return The denominator of q.
+   */
+  Integer denominator(const Rational& q) const;
+
+  /*! Convert an integer to an algebraic number.
+   * \param z An integer.
+   * \return The algebraic number equivalent to z.
+   */
+  Algebraic convert(const Integer& z) const;
+
+  /*! Convert a rational number to an algebraic number.
+   * \param q A rational number.
+   * \return The algebraic number equivalent to q.
+   */
+  Algebraic convert(const Rational& q) const;
+
+  /*! Construct a rational number that lies strictly between two algebraic
+   * values.
+   * \param x1 The first algebraic value.
+   * \param x2 The second algebraic value.
+   * \pre The two values are not equal.
+   * \return A rational number that lies in the open interval (x1, x2).
+   */
+  Rational rational_in_interval(const Algebraic& x1, const Algebraic& x2) const;
+
+  /*! Obtain a range of double-precision floats that contains the given
+   * algebraic number.
+   * \param x The given number.
+   * \return A pair <x_lo, x_hi> that contain x.
+   */
+  std::pair<double, double> double_interval(const Algebraic& x) const;
+
+  /*! Convert a sequence of rational coefficients to an equivalent sequence
+   * of integer coefficients. If the input coefficients are q(1), ..., q(k),
+   * where q(i) = n(i)/d(i) then the output coefficients will be of the
+   * form:
+   *               n(i) * lcm {d(1), ... , d(k)}
+   *       a(i) = -------------------------------
+   *               d(i) * gcd {n(1), ... , n(k)}
+   *
+   * \param q_begin The begin iterator of the rational sequence.
+   * \param q_end The past-the-end iterator of the rational sequence.
+   * \param zoi An output iterator for the integer coefficients.
+   * \return A past-the-end iterator for the output container.
+   * \pre The value type of q_begin and q_end is `Rational`, and
+   *      the value type of zoi is `Integer`.
```

Comment:
Backticks here should be  single quotes?
zoi, q_begin and q_end between backtics?